meta-facebook/yv35-cl/src/platform/plat_sensor_table.c

```diff
@@ -367,7 +387,33 @@ void pal_extend_sensor_config()
 		}
 	}
 
+	/* Follow the hardware design,
+	 * the GPIOA7(HSC_SET_EN_R) should be set to "H"
+	 * and the 2OU configuration is set if the 2OU is present.
+	 */
 	CARD_STATUS _2ou_status = get_2ou_status();
+	if (_2ou_status.present) {
+		sensor_count = ARRAY_SIZE(mp5990_sensor_config_table);
+		for (int index = 0; index < sensor_count; index++) {
+			mp5990_sensor_config_table[index].init_args = &mp5990_init_args[1];
+		}
+		sensor_count = ARRAY_SIZE(ltc4286_sensor_config_table);
+		for (int index = 0; index < sensor_count; index++) {
+			ltc4286_sensor_config_table[index].init_args = &ltc4286_init_args[1];
+		}
+		gpio_set(HSC_SET_EN_R, GPIO_HIGH);
+	} else {
+		sensor_count = ARRAY_SIZE(mp5990_sensor_config_table);
+		for (int index = 0; index < sensor_count; index++) {
+			mp5990_sensor_config_table[index].init_args = &mp5990_init_args[0];
+		}
+		sensor_count = ARRAY_SIZE(ltc4286_sensor_config_table);
+		for (int index = 0; index < sensor_count; index++) {
+			ltc4286_sensor_config_table[index].init_args = &ltc4286_init_args[0];
+		}
+		gpio_set(HSC_SET_EN_R, GPIO_LOW);
+	}
```

Comment:
These two branches look almost identical except for some parameters. Can this be refactored to less code.

Something like:
```
        int arg_index = (_2ou_status.present) ? 1 : 0;
        int gpio_state = (_2ou_status.present) ? GPIO_HIGH : GPIO_LOW;
	
	sensor_count = ARRAY_SIZE(mp5990_sensor_config_table);
	for (int index = 0; index < sensor_count; index++) {
		mp5990_sensor_config_table[index].init_args = &mp5990_init_args[arg_index];
	}
	sensor_count = ARRAY_SIZE(ltc4286_sensor_config_table);
	for (int index = 0; index < sensor_count; index++) {
		ltc4286_sensor_config_table[index].init_args = &ltc4286_init_args[arg_index];
	}
	gpio_set(HSC_SET_EN_R, gpio_state);
```

Reply:
Thanks for your suggestion.
The code is forced updated.
Please review it again.